xilem_web/src/after_update.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2023 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use xilem_core::{MessageResult, Mut, View, ViewId, ViewMarker};
+
+use crate::{DomNode, DomView, DynMessage, ViewCtx};
+
+pub struct AfterBuild<E, F> {
+    element: E,
+    callback: F,
+}
+
+pub struct AfterRebuild<E, F> {
+    element: E,
+    callback: F,
+}
+
+pub struct BeforeTeardown<E, F> {
+    element: E,
+    callback: F,
+}
+
+/// Invokes the `callback` after the inner `element` [`DomView`] was created.
+/// The callback has a reference to the raw DOM node as its only parameter.
+///
+/// Caution: At this point, however,
+/// no properties have been applied to the node.
+///
+/// The use of this function should be avoided and
+/// should only be utilized in exceptional cases!
+pub fn after_build<E, F>(element: E, callback: F) -> AfterBuild<E, F> {
+    AfterBuild { element, callback }
+}
+
+/// Invokes the `callback` after the inner `element` [`DomView<State>`]
+/// was rebuild, which usually happens after anything has changed in the `State` .
+///
+/// Memoization can prevent `callback` being called.
+/// The callback has a reference to the raw DOM node as its only parameter.
+///
+/// The use of this function should be avoided and
+/// should only be utilized in exceptional cases!
+pub fn after_rebuild<E, F>(element: E, callback: F) -> AfterRebuild<E, F> {
+    AfterRebuild { element, callback }
+}
+
+/// Invokes the `callback` before the inner `element` [`DomView`] (and its underlying DOM node) is destroyed.
+/// As accessing the underlying raw DOM node can mess with the inner logic of `xilem_web`,
+/// this should only be used as an escape-hatch for properties not supported by `xilem_web`.
+/// E.g. to be interoperable with external javascript libraries.
+pub fn before_teardown<E, F>(element: E, callback: F) -> BeforeTeardown<E, F> {
+    BeforeTeardown { element, callback }
+}
+
+impl<E, F> ViewMarker for AfterBuild<E, F> {}
+impl<E, F> ViewMarker for AfterRebuild<E, F> {}
+impl<E, F> ViewMarker for BeforeTeardown<E, F> {}
+
+impl<State, V, F> View<State, (), ViewCtx, DynMessage> for AfterBuild<V, F>
+where
+    State: 'static,
```

Comment:
Likely needs a bound then as well
```suggestion
    State: 'static,
    Action: 'static,
```